API_Assignment

NawaphonB opened the pull request.

NawaphonB committed:
Applying yaml file as a testdata in and beautify coding

massarinbdoppio commented on Testcase/Testcase.robot:
```diff
@@ -29,41 +25,37 @@ TC-004 Verify that create asset API can work correctly
-
+    Call Create Asset API   ${TC_004.asset_id}    ${TC_004.asset_name}   ${TC_004.asset_type}   ${TC_004.asset_inuse}
+    Verified result by calling get API      ${TC_004.asset_id}    ${TC_004.asset_name}   ${TC_004.asset_type}   ${TC_004.asset_inuse}
+    Reset Data by delete asset      ${TC_004.asset_id}
```
อ่า อันนี้ดีที่คิดมาค่ะ
ทีนี้ มันควรอยู่ใน [Teardown] ค่ะ รู้ไหมนะคะ ว่าทำไม
**[Teardown]** Reset Data by delete asset      ${TC_004.asset_id}

massarinbdoppio commented on Testcase/Testcase.robot:
```diff
@@ -29,41 +25,37 @@ TC-004 Verify that create asset API can work correctly
+    Verified result by calling get API      ${TC_006.asset_id}    ${TC_006.asset_name}    ${TC_006.asset_type}   ${TC_006.asset_inuse}
+    Call Modify Asset API   ${TC_006.asset_id_mod}    ${TC_006.asset_name_mod}    ${TC_006.asset_type_mod}   ${TC_006.asset_inuse_mod}
+    Verified result by calling get API      ${TC_006.asset_id_mod}    ${TC_006.asset_name_mod}    ${TC_006.asset_type_mod}    ${TC_006.asset_inuse_mod}
+    Reset Data by delete asset      ${TC_006.asset_id_mod}
```
มันควรอยู่ใน [Teardown]

massarinbdoppio commented on Testcase/.pabotsuitenames:
```diff
@@ -1,5 +1,5 @@
-datasources:401db91b9834e657110d181fe200050bd3fed4fd
+datasources:9ceb4104a6b3f53967d97fa4cb6ca573ec315cad
```
อันนี้ไม่เอาขึ้น git ค่ะ
เป็น result ของ pabot

massarinbdoppio commented on Keyword/Verified_Keyword.robot:
```diff
@@ -0,0 +1,29 @@
+*** Keywords ***
```
อันนี้ ควรอยู่ใน แต่ละ feature ของ api ค่ะ ไม่ควรมารวมในไฟล์ที่ชื่อ verified เพราะกลายเป็นว่า ทุก verified fail error จะมากองอยู่ในนี้ ซึ่งจิงๆ มันควรอยู่ใน feature ของแต่ละอันมากกว่า
เช่น Call API Login & Verified Error Message TC_001  >> api login -- ก็ควรอยู่ใน api_list.Login หนะ
Call API Get asset & Verified Error Message TC_003 >> api get เหมือนกัน

massarinbdoppio commented on Keyword/common_Keyword.robot:
```diff
@@ -1,7 +1,7 @@
 *** Keywords ***
 Login & Get Token 
```
อันนี้ควรอยู่ใน Login ค่ะ เพราะมันคือ login success

massarinbdoppio commented on Keyword/API_list/Get_Asset.robot:
```diff
@@ -0,0 +1,26 @@
+*** Keywords ***
+Verified delete result by calling get API
```
ชื่อนี้พอใช้ได้นะคะ  เนื่องจากเคสของนาวมันตรงๆ ตัวว่า ลบทิ้งต้องไม่ได้ค่ามา

ทีนี้ ให้ชื่อเพิ่มไปว่าตั้งกลางๆ แบบนี้ก็ได้ค่ะ เพื่อรองรับการใช้งานที่มากขึ้นกว่าเทสเคสที่ทำตอนนี้
เพราะการ delete คือเช็คว่า get แล้วต้องไม่มีค่ากลับมาถูกป่ะ 
ชื่อ Verify get api response without asset id
ทีนี้มันก็สามารถเอาไปใช้ในเทสเคสกรณีที่เทสว่า get ด้วย asset id ที่ไม่มีอยู่จริงๆ แล้วก็ต้องไม่ได้ค่าออกมาเหมือนกัน

massarinbdoppio commented on Keyword/API_list/Get_Asset.robot:
```diff
@@ -0,0 +1,26 @@
+    ${morethanone}=     Evaluate    ${count}>1
+    Should Be True      ${morethanone}
+
+Verified result by calling get API
```
จาก comment ข้างบน ชื่อนี้ถ้าจะเปลี่ยนให้ไปทางเดียวกัน
Verify get api response with asset id

NawaphonB committed:
Feedback PR

-Reformat keyword file
-Rename keyword make it more flexible
-Adding Reference to every keyword
